cmd/podman/containers/cp.go

```diff
@@ -382,13 +382,16 @@ func copyToContainer(container string, containerPath string, hostPath string) er
 		if err != nil {
 			return err
 		}
+		defer os.Remove(tmpFile.Name())
+
 		_, err = io.Copy(tmpFile, os.Stdin)
 		if err != nil {
 			return err
 		}
 		if err = tmpFile.Close(); err != nil {
```

Comment:
This shoould be deferred also.

Reply:
I think it is better if we close immediately after we finish writing to it, but we must make sure it is also closed in the error path of `io.Copy`.

Reply:
opened a PR to close the file on `io.Copy` errors: https://github.com/containers/podman/pull/19534